docs/examples/secrets.md

```diff
@@ -37,3 +37,103 @@ print(model.model_dump())
 print(model.model_dump_json())
 #> {"password":"IAmSensitive","password_bytes":"IAmSensitiveBytes"}
 ```
+
+## Create your own Secret field
+
+Pydantic provides the generic `Secret` class as a mechanism for creating custom secret types.
+
+??? api "API Documentation"
+    [`pydantic.types.Secret`][pydantic.types.Secret]<br>
+
+Pydantic provides the generic `Secret` class as a mechanism for creating custom secret types.
+You can either directly parametrize `Secret`, or subclass from a parametrized `Secret` to customize the `str()` and `repr()` of a secret type.
+
+```py
+from datetime import date
+
+from pydantic import BaseModel, Secret
+
+# Using the default representation
+SecretDate = Secret[date]
+
+
+# Overwriting the representation
+class SecretSalary(Secret[float]):
+    def _display(self) -> str:
+        return '$****.**'
+
+
+class Employee(BaseModel):
+    date_of_birth: SecretDate
+    salary: SecretSalary
+
+
+employee = Employee(date_of_birth='1990-01-01', salary=42)
+
+print(employee)
+#> date_of_birth=Secret('**********') salary=SecretSalary('$****.**')
+
+print(employee.salary)
+#> $****.**
+
+print(employee.salary.get_secret_value())
+#> 42.0
+
+print(employee.date_of_birth)
+#> **********
+
+print(employee.date_of_birth.get_secret_value())
+#> 1990-01-01
+```
+
+Pydantic currently supports enforcing constraints on the underlying type of a secret type, but not the secret type itself.
```

Comment:
```suggestion
You can enforce constraints on the underlying type through annotations:
```
The way it's currently phrased, it sounds like there is some capability here, but I don't think it necessarily makes sense to enforce constraints on "the secret type itself".

I think phrasing it like this makes it more clear that "this is the right way to do the thing you're thinking of", as opposed to, "this is the way to do a thing, which may or may not be what you are thinking of".